src/servicebus/Elsa.AzureServiceBus/MessageHandlers/SignalMessageHandler.cs

```diff
@@ -0,0 +1,109 @@
+﻿using Elsa.Models;
+using Elsa.Persistence;
+using Elsa.Services;
+using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Linq;
+using Newtonsoft.Json.Linq;
+using Elsa.AzureServiceBus.Activities;
+using Elsa.AzureServiceBus.Models;
+using Elsa.AzureServiceBus.Notifications;
+using Elsa.AzureServiceBus.Services;
+using Elsa.Services.Models;
+
+namespace Elsa.AzureServiceBus.MessageHandlers
+{
+    /// <summary>
+    /// Processes any incoming signal messages received via servicebus 
+    /// </summary>
+    public class SignalMessageHandler : INotificationHandler<ProcessMessageNotification>
+    {
+        private readonly ITokenService _tokenService;
+        private readonly IServiceProvider _serviceProvider;
+
+        public SignalMessageHandler(IServiceProvider serviceProvider, ITokenService tokenService)
+        {
+            _serviceProvider = serviceProvider;
+            _tokenService = tokenService;
+        }
+
+        public async Task Handle(ProcessMessageNotification notification, CancellationToken cancellationToken)
+        {
```

Comment:
This method has a lot of nested code blocks. Please invert the if statements and exit early; it will make it a easier to read the contents of the method.